src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,284 @@
+                "org.apache.spark.api.java.JavaSparkContext",
+                "fromSparkContext",
+                sc);
+
+            Version version = SparkEnvironment.SparkVersion;
+            return (version.Major, version.Minor) switch
+            {
+                (2, 3) when version.Build == 0 || version.Build == 1 =>
+                    CreateBroadcast_V2_3_1_AndBelow(javaSparkContext, value),
+                (2, 3) => CreateBroadcast_V2_3_2_AndAbove(javaSparkContext, sc, value),
+                (2, 4) => CreateBroadcast_V2_3_2_AndAbove(javaSparkContext, sc, value),
+                _ => throw new NotSupportedException($"Spark {version} not supported.")
+            };
+        }
+
+        /// <summary>
+        /// Calls the necessary functions to create org.apache.spark.broadcast.Broadcast object
+        /// for Spark versions 2.3.0 and 2.3.1 and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="javaSparkContext">Java Spark context object</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_1_AndBelow(
+            JvmObjectReference javaSparkContext,
+            object value)
+        {
+            WriteToFile(value);
+            return (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "readBroadcastFromFile",
+                javaSparkContext,
+                _path);
+        }
+
+        /// <summary>
+        /// Calls the necessary Spark functions to create org.apache.spark.broadcast.Broadcast
+        /// object for Spark versions 2.3.2 and above, and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="javaSparkContext">Java Spark context object</param>
+        /// <param name="sc">SparkContext object</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_2_AndAbove(
+            JvmObjectReference javaSparkContext,
+            SparkContext sc,
+            object value)
+        {
+            bool encryptionEnabled = bool.Parse(
+                sc.GetConf().Get("spark.io.encryption.enabled", "false"));
+
+            if (encryptionEnabled)
+            {
+                throw new NotImplementedException("Broadcast encryption is not supported yet.");
+            }
+            else
+            {
+                WriteToFile(value);
+            }
+
+            var pythonBroadcast = (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "setupBroadcast",
+                _path);
+
+            return (JvmObjectReference)javaSparkContext.Jvm.CallNonStaticJavaMethod(
```

Comment:
Can't you just do `javaSparkContext.Invoke`?

Reply:
Yes, it's calling the function on `javaSparkContext` itself so should have done that. Thanks for pointing out! Made the change